### PR TITLE
Amb 1749 create unit tests for validations 42 to 60

### PR DESCRIPTION
**Pull request by nathanwall**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: lambda_code/src/models/fhir_immunization.py [hunk 1]
```diff
@@ -754,6 +754,389 @@ def pre_validate_reason_code_coding_displays(cls, values: dict) -> dict:
...
+        LOCAL_PATIENT_ID) exists, then it is a non-empty string
+        """
+        try:
+            local_patient_code = get_generic_questionnaire_response_value(
+                values, "LocalPatient", "code"
+            )
+            pre_validate_string(
+                local_patient_code,
+                generate_field_location_for_questionnnaire_response(
+                    link_id="LocalPatient", field_type="code"
```
Comment (AlexandraBenson) [inline lambda_code/src/models/fhir_immunization.py:837]: Also needs to handle 'max 20 characters rule' (see word document FHIR API spec)

File: lambda_code/src/models/fhir_immunization.py [hunk 2]
```diff
@@ -754,6 +754,389 @@ def pre_validate_reason_code_coding_displays(cls, values: dict) -> dict:
...
+            pass
+
+        return values
+
+    @classmethod
+    def pre_validate_submitted_time_stamp_code(cls, values: dict) -> dict:
+        """
+        Pre-validate that, if contained[?(@.resourceType=='QuestionnaireResponse')]
+        .item[?(@.linkId=='SubmittedTimeStamp')].answer[0].valueCoding.code (legacy CSV field name:
+        SUBMITTED_TIMESTAMP) exists, then it is a non-empty string
```
Comment (AlexandraBenson) [inline lambda_code/src/models/fhir_immunization.py:1048]: Doc string should reflect checks for datetime, not just generic string (can be copied from another datetime validator e.g. occurfence_date_time)

File: lambda_code/tests/test_immunization_pre_validator_model.py [hunk 3]
```diff
@@ -514,3 +522,210 @@ def test_model_pre_validate_reason_code_coding_displays(self):
...
+
+    def test_model_pre_validate_reduce_validation_code(self):
+        """
+        Test pre_validate_reduce_validation_code accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
+            self,
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='ReduceValidation')].answer[0].valueCoding.code",
+            valid_strings_to_test=["True", "False"],
```
Comment (AlexandraBenson) [inline lambda_code/tests/test_immunization_pre_validator_model.py:719]: Needs to test for the enums rule (use arguments predefined_values = ("True", "False") and invalid_strings_to_test=....[insert list of some invalid non-empty strings here]).

File: lambda_code/tests/test_immunization_pre_validator_model.py [hunk 4]
```diff
@@ -514,3 +522,210 @@ def test_model_pre_validate_reason_code_coding_displays(self):
...
+            field_location="contained[?(@.resourceType=='QuestionnaireResponse')]"
+            + ".item[?(@.linkId=='SiteCode')].answer[0].valueCoding.system",
+            valid_strings_to_test=["snomed"],
+        )
+
+    def test_model_pre_validate_local_patient_code(self):
+        """
+        Test pre_validate_local_patient_code accepts valid values and rejects invalid values
+        """
+        ValidatorModelTests.test_string_value(
```
Comment (AlexandraBenson) [inline lambda_code/tests/test_immunization_pre_validator_model.py:563]: Needs to test for max 20 characters

File: lambda_code/src/models/fhir_immunization.py (hunk 1 above)
Comment (nathanwall) [reply lambda_code/src/models/fhir_immunization.py:837]: done

File: lambda_code/src/models/fhir_immunization.py (hunk 2 above)
Comment (nathanwall) [reply lambda_code/src/models/fhir_immunization.py]: done

File: lambda_code/tests/test_immunization_pre_validator_model.py (hunk 3 above)
Comment (nathanwall) [reply lambda_code/tests/test_immunization_pre_validator_model.py:721]: done

File: lambda_code/tests/test_immunization_pre_validator_model.py (hunk 4 above)
Comment (nathanwall) [reply lambda_code/tests/test_immunization_pre_validator_model.py:563]: done

**Review by AlexandraBenson — APPROVED** [review]

**Review by lee-richardson-nhs — APPROVED** [review]

**Pull request merged**